### PR TITLE
usagereporter: Use the batched event ingest RPC

### PR DESCRIPTION
**Pull request by espadolini**
Depends on gravitational/cloud#3716

This PR switches the `usagereporter` submitter to the new `SubmitEvents` RPC and bumps the limits for minimum and maximum batch size to values that are still somewhat conservative.

### CODE REVIEW

File: lib/usagereporter/teleport/usagereporter.go [hunk 1]
```diff
@@ -181,21 +181,19 @@ func NewPrehogSubmitter(ctx context.Context, prehogEndpoint string, clientCert *
...
-				errors = append(errors, err)
-			}
+		evs := make([]*prehogv1.SubmitEventRequest, 0, len(events))
+		for _, e := range events {
+			evs = append(evs, e.Event)
 		}
 
-		return failed, trace.NewAggregate(errors...)
+		req := connect.NewRequest(&prehogv1.SubmitEventsRequest{
+			Events: evs,
```
[the remark this answers is not in the document]
Comment (espadolini) [reply lib/usagereporter/teleport/usagereporter.go:190]: I'm not sure if the responsibility of that should fall to the submitter function - the `UsageReporter` will internally enforce the `MaxBatchSize` limit passed in at creation time.

[the remark this answers is not in the document]
Comment (espadolini) [reply lib/usagereporter/teleport/usagereporter.go:190]: I expect the limit of 500 events to be well under the transport limits (by a factor of at least 5) but if we actually happen to go over then we'll just attempt to put the events back in the queue until each individual event hits its own retry limit. Do you think it's worth it to also add some halving logic in the call to `SubmitEvents` here, maybe doing a single successful RPC with a subset of the events and returning the other events as "failed", which is a thing that the event submitter API supports?

[the remark this answers is not in the document]
Comment (espadolini) [reply lib/usagereporter/teleport/usagereporter.go:190]: I think that handling that sort of partial failure correctly will require a bit of work on the logic of the usagereporter, and until this code ends up being used for on-prem we're not going to have problems with that sort of breakage.

Unless you feel strongly about this I'd rather just leave it as is.

**Review by rosstimothy — APPROVED** [review]

**Comment by espadolini** [comment]
@Joerger friendly ping

File: proto/prehog/v1alpha/teleport.proto [hunk 2]
```diff
@@ -381,15 +381,22 @@ message SubmitEventRequest {
...
+message SubmitEventsRequest {
+  repeated SubmitEventRequest events = 1;
+}
+message SubmitEventsResponse {}
+
 message HelloTeleportRequest {}
 message HelloTeleportResponse {}
 
 service TeleportReportingService {
   rpc SubmitEvent(SubmitEventRequest) returns (SubmitEventResponse) {}
```
Comment (Joerger) [inline proto/prehog/v1alpha/teleport.proto:399]: Are we still using SubmitEvent, or should we deprecate it?

**Review by Joerger — APPROVED** [review]

File: proto/prehog/v1alpha/teleport.proto (hunk 2 above)
Comment (espadolini) [reply proto/prehog/v1alpha/teleport.proto:399]: Good point, we're not using `SubmitEvent` anymore (and if we needed a single event submit we could just send a single event through `SubmitEvents` anyway).

I added a deprecation notice in gravitational/cloud#3767, it'll get in here the next time we sync the proto.

**Pull request merged**